haystack/components/generators/utils.py

```diff
@@ -112,8 +112,13 @@ def _convert_streaming_chunks_to_chat_message(chunks: List[StreamingChunk]) -> C
             )
 
     # finish_reason can appear in different places so we look for the last one
+    # First check the dedicated finish_reason field, then fall back to meta for backward compatibility
+    # NOTE: This fallback is required during migration period until all chat generators
+    # are updated to populate the dedicated StreamingChunk.finish_reason field.
     finish_reasons = [
-        chunk.meta.get("finish_reason") for chunk in chunks if chunk.meta.get("finish_reason") is not None
+        chunk.finish_reason or chunk.meta.get("finish_reason")
+        for chunk in chunks
+        if chunk.finish_reason or chunk.meta.get("finish_reason")
```

Comment:
We don't need the fallback here. Instead let's update the Generators and ChatGenerators that use this method which should just be the ones within Haystack main. E.g. `OpenAIGenerator` and then the HuggingFace ones too. 

No integrations use this method yet since it was only recently introduced so no need for the fallback or warning message.